ICAFIX/scripts/functionhighpassandvariancenormalize.m

```diff
@@ -32,94 +34,125 @@ function functionhighpassandvariancenormalize(TR,hp,fmri,WBC,varargin)
-  dovol = 0; %regname is only used for a new surface registration, so will never require redoing volume
-  regstring = varargin{1}; %this has the underscore on the front already
-  if ~ischar(regstring)
-	error('%s: REGSTRING should be a string', mfilename);
-  end
+    dovol = false; %regname is only used for a new surface registration, so will never require redoing volume
+    regstring = varargin{1}; %this has the underscore on the front already
+    if ~ischar(regstring)
+        error('%s: REGSTRING should be a string', mfilename);
+    end
 end
 
 if length(varargin) >= 2
-  if isdeployed()
-    volhpwishart = str2double(varargin{2});
-  else
-    volhpwishart = varargin{2};
-  end
-end   
-
-% Check whether polynomial detrend is being requested for the high-pass filtering.
-if ischar(hp)
-  hp = lower(hp);
-  if strncmp(hp,pdstring,numel(pdstring))
-	pdflag = true;
-	hp = str2double(hp(numel(pdstring)+1:end));  % hp is now a numeric representing the order of the polynomial detrend
-  elseif ~isempty(str2double(hp))  % Allow for hp to be provided as a string that contains purely numeric elements
-	hp = str2double(hp);
-  else error('%s: Invalid specification for the high-pass filter', mfilename);
-  end
+    if isdeployed()
+        volwishart = str2double(varargin{2});
+    else
+        volwishart = varargin{2};
+    end
 end
 
-%% Allow for compiled matlab (hp as a string is already handled above)
-if (isdeployed)
-  TR = str2double(TR);
+if length(varargin) >= 3
+    if isdeployed()
+        ciftiwishart = str2double(varargin{3});
+    else
+        ciftiwishart = varargin{3};
+    end
 end
 
-%% Argument checking
-if hp==0 && ~pdflag
-  warning('%s: hp=0 will be interpreted as polynomial detrending of order 1', mfilename);
-  pdflag=true;
-  hp=1;
-  % But in this case, preserve use of "_hp0" in the filename to indicate a linear detrend (for backwards compatibility in file naming)
-  hpstring = '_hp0';
+if length(varargin) >= 4
+    switch varargin{4}
+        case {'default', ''}
+            %leave things alone
+        case 'melodiclike'
+            iters = 1;
+            VNhalfdim = true;
+        otherwise
+            error(['unknown ICADIMMODE value: ' varargin{4}]);
+    end
 end
-if pdflag
-  if (~isscalar(hp) || hp < 0 || hp ~= fix(hp))
-	error('%s: Invalid specification for the order of the polynomial detrending', mfilename);
-  end
+
+%% Allow for compiled matlab
+if isdeployed()
+    TR = str2double(TR);
 end
 
-%% Finish setting up hpstring to use in file names
-if ~pdflag
-  pdstring = '';
+% Check whether polynomial detrend is being requested for the high-pass filtering.
+% Coincidentally deals with compiled matlab arguments
+if ischar(hp)
+    hp = lower(hp);
+    if strncmp(hp,pdstring,numel(pdstring))
+        pdflag = true;
+        polydeg = str2double(hp(numel(pdstring)+1:end));  % different purposes should use a different variable name
+        if (~isscalar(hp) || hp < 0 || hp ~= fix(hp))
```

Comment:
Should this be using `polydeg` rather than `hp`?

Reply:
probably